sbt-dumpschema-plugin/src/main/scala/org/virtuslab/akkasaferserializer/SchemaWriter.scala

```diff
@@ -0,0 +1,43 @@
+package org.virtuslab.akkasaferserializer
+
+import better.files.File
+import io.bullet.borer.Json
+import org.virtuslab.akkasaferserializer.model.{Codecs, TypeDefinition}
+
+import scala.collection.mutable
+
+class SchemaWriter(outputDirectory: File) extends Codecs {
+
+  def this(dumpSchemaOptions: DumpSchemaOptions) = {
+    this(File(dumpSchemaOptions.outputDir))
+  }
+
+  lazy val lastDump: Map[String, TypeDefinition] = {
+    outputDirectory
+      .createDirectoryIfNotExists(createParents = true)
+      .list(_.extension.contains(".json"))
+      .flatMap { file =>
+        Json.decode(file.byteArray).to[TypeDefinition].valueTry.toOption
+      }
+      .map(x => (x.name, x))
+      .toMap
+  }
+
+  private val dumped: mutable.Set[String] = mutable.Set()
+
+  def isUpToDate(name: String): Boolean = dumped(name)
+  def offerDump(typeDefinition: TypeDefinition): Unit = {
+    if (lastDump.get(typeDefinition.name).fold(true)(_ != typeDefinition)) {
+      dump(typeDefinition)
+    }
+    dumped += typeDefinition.name
+  }
+
+  private def dump(typeDefinition: TypeDefinition) = {
+    (outputDirectory / (typeDefinition.name + ".json"))
+      .createIfNotExists()
```

Comment:
Hmmm so schema for each type goes into a separate file? For the sake of comparing the entire schemas, it might make more sense to pack all the types into a biiiig single file instead 🤔